Description: Replaces fvwm's AM_CHECK_PKG_CONFIG with PKG_PROG_PKG_CONFIG.
  Use upstream PKG_PROG_PKG_CONFIG instead of fvwm's AM_CHECK_PKG_CONFIG
  macro for pkg-config. This should improve cross building.
  .
  Upstream was not interested in the patch.
Author: Helmut Grohne <helmutg@debian.org>
Debian-Bug: http://bugs.debian.org/900074
Forwarded: not-needed
Reviewd-By: Jaimos Skriletz <jaimosskriletz@gmail.com>
Last-Updated: 2018-06-12
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -542,37 +542,6 @@ c = locale_charset ();
 ])
 
 #-----------------------------------------------------------------------------
-# pkg-config
-
-dnl
-dnl
-AC_DEFUN([AM_CHECK_PKG_CONFIG],
-[dnl
-dnl Get the cflags and libraries from the freetype-config script
-dnl
-AC_ARG_WITH(pkgconfig-prefix,
-AS_HELP_STRING([--with-pkgconfig-prefix=PFX],[prefix where pkg-config is installed]),
-            pkgconfig_config_prefix="$withval", pkgconfig_config_prefix="")
-AC_ARG_WITH(pkgconfig-exec-prefix,
-AS_HELP_STRING([--with-pkgconfig-exec-prefix=PFX],[exec prefix where pkg-config is installed]),
-            pkgconfig_config_exec_prefix="$withval",pkgconfig_config_exec_prefix="")
-
-if test x$pkgconfig_config_exec_prefix != x ; then
-  pkgconfig_config_args="$pkgconfig_config_args --exec-prefix=$pkgconfig_config_exec_prefix"
-  if test x${PKG_CONFIG+set} != xset ; then
-    PKG_CONFIG=$pkgconfig_config_exec_prefix/bin/pkg-config
-  fi
-fi
-if test x$pkgconfig_config_prefix != x ; then
-  pkgconfig_config_args="$pkgconfig_config_args --prefix=$pkgconfig_config_prefix"
-  if test x${PKG_CONFIG+set} != xset ; then
-    PKG_CONFIG=$pkgconfig_config_prefix/bin/pkg-config
-  fi
-fi
-AC_PATH_PROG(PKG_CONFIG, pkg-config, no)
-])
-
-#-----------------------------------------------------------------------------
 # Configure paths for fontconfig
 # Marcelo Magallon 2001-10-26, based on gtk.m4 by Owen Taylor
 # modified by olicha for fontconfig
--- a/configure.ac
+++ b/configure.ac
@@ -308,7 +308,7 @@ esac
 # unfortunately, we need pkg-config for the detection of certain libs:
 # - version of fontconfig without fontconfig-config
 # - version of fribidi without fribidi-config
-AM_CHECK_PKG_CONFIG
+PKG_PROG_PKG_CONFIG
 
 PKG_CHECK_MODULES([xt],[xt],
 [],
@@ -772,7 +772,7 @@ AC_ARG_ENABLE(rsvg,
 )
 if test ! x"$with_rsvg" = xno; then
    with_rsvg=no
-   if test ! x"$PKG_CONFIG" = xno ; then
+   if test ! x"$PKG_CONFIG" = x ; then
       AC_MSG_CHECKING(for librsvg - version >= $rsvg_min_version)
       if $PKG_CONFIG --exists librsvg-2.0 ; then
          if $PKG_CONFIG --exists "librsvg-2.0 >= $rsvg_min_version" ; then
@@ -1021,7 +1021,7 @@ if test ! x"$with_bidi" = xno; then
     AS_HELP_STRING([--with-fribidi-bindir=DIR],
       [directory of fribidi-config if not in PATH]),
     FRIBIDI_BINDIR="$withval", FRIBIDI_BINDIR=".")
-  if test ! x"$PKG_CONFIG" = xno && $PKG_CONFIG --exists "fribidi >= $fribidi_min_version"; then
+  if test ! x"$PKG_CONFIG" = x && $PKG_CONFIG --exists "fribidi >= $fribidi_min_version"; then
     FRIBIDI_CONFIG="$PKG_CONFIG fribidi"
   else
     AC_PATH_PROG(FRIBIDI_CONFIG, fribidi-config,, [$FRIBIDI_BINDIR:$PATH])
